### ISSUE TITLE
File structure in `build` directory

### ISSUE THREAD

**Issue by mssun** [mesalock-linux/mkpkg#1]
We should have a better file structure in the `build` directory like this

```
└── build
    └── linux-4.16.14              # the name should be ${pkgname}-${pkgver}
        ├── linux-4.16.14.tar.xz   # package
        ├── log                           # ${logdir}
        │   ├── stderr.out
        │   └── stdout.out
        ├── pkg                           # ${pkgdir}
        └── src                           # ${srcdir}
            ├── linux-4.16.14
            │   ├── cloned_files
            │   └── extracted_files
            ├── linux-4.16.14.tar.gz     # downloaded zip file
            └── other_files
```

The working directory of `prepare`, `build` and `install` should be `${srcdir}`.

The `${logdir}`, `${pkgdir}`, `${srcdir}` environmental variables should be also set as absolute path when `prepare`, `build` and `install` the package

**Comment by Arcterus** [comment]
I don't think `${srcdir}` should be defined as the working directory.  The working directories of `prepare` and `build`/`install` are different at the moment (the former uses `linux-4.16.14/src` while the latter uses `linux-4.16.14/build`) due to the possibility of `skip_extract`.  Maybe `${srcdir}` should be `linux-4.16.14/src` while `${blddir}` or `${builddir}` should be `linux-4.16.14/build`?

**Comment by mssun** [comment]
I agree with you.

**Comment by Arcterus** [comment]
This should be resolved by 0417ed030db98a03ea3d7c2486040face4e3b599 and its parent.
